data-structures-queues/take-2nd.js

```diff
@@ -0,0 +1,15 @@
+/* exported take2nd */
+
+function take2nd(queue) {
+  const front = queue.peek();
+  if (front !== undefined) {
+    const taken = queue.dequeue(front);
+    if (queue.peek() !== undefined) {
+      const second = queue.dequeue();
+      queue.enqueue(taken);
+      return second;
+    } else {
+      return taken;
+    }
+  }
```

Comment:
Try to simplify this a bit. `.dequeue()` doesn't take any arguments so it's not clear what is going on here.